src/ansi-c/parser.y

```diff
@@ -532,15 +532,62 @@ target_list:
         ;
 
 target:
+          deref_target
+        | target '[' array_index ']'
+        { binary($$, $1, $2, ID_range, $3); }
+        | target '[' array_range ']'
+        { binary($$, $1, $2, ID_range, $3); }
+        ;
+
+array_index:
           identifier
-        | '*' target
+        | integer
+        ;
+
+array_range:
```

Comment:
Not a blocker, but a comment from me: I'm quite worried about the proliferation of nonstandard extensions. We're getting into a situation where we have to support gcc extensions, clang extensions, visual studio extensions on top of our own modifications to the language...

This could've been done without any parser changes at the relatively small cost of a less cute syntax, e.g. `__CPROVER_array_range(array, lower, upper)` or something like that. Looks bad, but less problematic syntactically.

Reply:
Based on the discussion on #5966 I reverted all the parser changes (to support array ranges) from this PR. We should have a syntax similar to ACSL, but that should be added on a separate PR. I kept the code on the assigns clause size for this future support.